src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -538,3 +542,56 @@ export const updateFormSettings = (
     return okAsync(updatedForm.getSettings())
   })
 }
+
+/**
+ * Deletes form logic.
+ * @param form The original form to delete logic in
+ * @param logicId the logicId to delete
+ * @returns ok(true) on success
+ * @returns err(database errors) if db error is thrown during logic delete
+ * @returns err(LogicNotFoundError) if logicId does not exist on form
+ */
+export const deleteFormLogic = (
+  form: IPopulatedForm,
+  logicId: string,
+): ResultAsync<true, DatabaseError | LogicNotFoundError> => {
+  // First check if specified logic exists
+  if (!form.form_logics.some((logic) => logic.id === logicId)) {
+    logger.error({
+      message: 'Error occurred when deleting form logic',
+      meta: {
+        action: 'deleteFormLogic',
+        formId: form._id,
+        logicId,
+      },
+    })
+    return errAsync(new LogicNotFoundError())
+  }
+
+  // Remove specified logic and then update form logic
+  return ResultAsync.fromPromise(
+    FormModel.findByIdAndUpdate(
+      form._id,
+      {
+        $pull: { form_logics: { _id: logicId } },
+      },
+      {
+        new: true,
+        runValidators: true,
+      },
+    ).exec(),
+    (error) => {
+      logger.error({
+        message: 'Error occurred when deleting form logic',
+        meta: {
+          action: 'deleteFormLogic',
+          formId: form._id,
+          logicId,
+        },
+        error,
+      })
+      return transformMongoError(error)
+    },
+    // On success, return true
+  ).map(() => true)
```

Comment:
instead of throwing away information by returning `true`, might it be better to return the saved form so the caller can decide what to do with it?

Reply:
since we are deleting logic here, i thought it would suffice to return true since there is no new information created from the API Call (similar to archive form)

Reply:
I guess, but then if the caller wanted to use the new form document, they would have to re-derive the new version by ownself deleting the relevant logic ID right? wouldn't it be easier to just pass back the updated document?

I know returning `true` is a pattern we use quite often but I don't think it's a good practice, except if there genuinely isn't any information to return

Reply:
edited